llama_index/agent/react/base.py

```diff
@@ -336,27 +364,47 @@ def stream_chat(
             chat_stream = self._llm.stream_chat(input_chat)
 
             # iterate over stream, break out if is final answer after the "Answer: "
-            is_done = False
             full_response = ChatResponse(
                 message=ChatMessage(content=None, role="assistant")
             )
-            for r in chat_stream:
-                if "Answer: " in (r.message.content or ""):
-                    is_done = True
-                    break
-                full_response = r
-            if is_done:
-                break
+            for latest_chunk in chat_stream:
+                latest_content = latest_chunk.message.content
+                if latest_content:
+                    if not latest_content.startswith(
+                        "Thought"
+                    ):  # doesn't follow thought-action format
+                        is_done = True
+                        full_response = latest_chunk
+                        break
+                    else:
+                        if "Answer: " in latest_content:
+                            is_done = True
+                            full_response = latest_chunk
+                            break
+                    full_response = latest_chunk
 
             # given react prompt outputs, call tools or return response
             reasoning_steps, _ = self._process_actions(
-                tools=tools, output=full_response
+                tools=tools, output=full_response, is_streaming=True
```

Comment:
I don't quite get what propagating `is_streaming` into the various function is doing right now. Could you explain a bit.

Reply:
Yeah, so this `is_streaming` needs to propagate a couple of functions to finally reach `output_parser.parse` which then it gets passed to its final destination, namely `ResponseReasoningStep`. Here, I made the distinction between `is_streaming` or not, in order to print either: `Response` for regular chat or `Response (starts with)` for stream chat. 😅

Reply:
very nit: could we just skip propagating the flag through output parser etc and just modify `ResponseReasoningStep` at higher level in the call stack? 

actually I'm not even sure how much cleaner it would be. Free free to ignore this.

Reply:
Ya, I'm not sure if it will be cleaner. Another option I thought of though equally messy (I believe) is to define a `StreamingResponseStep`, but I think a flag would still need to propagate up all the way to `output_parser`.